### PR TITLE
PN-148: Added age-of-onset and mode-of-inheritance to matching displa…

### PR DESCRIPTION
**Pull request by itaiGershtansky**
…y on patient page.

### CODE REVIEW

**Comment by itaiGershtansky** [comment]
match1

**Comment by itaiGershtansky** [comment]
@buske @veronikaslc please review

**Comment by buske** [comment]
This still displays the mode of inheritance section for me even when there's no data to display: 
screenshot

File: similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/AbstractPatientSimilarityView.java [hunk 1]
```diff
@@ -181,6 +188,28 @@ public String getContactToken()
         return this.contactToken;
     }
 
+    private String getAgeOfOnset(PatientData<List<SolrVocabularyTerm>> globalControllers) {
+        if (globalControllers != null) {
+            List<SolrVocabularyTerm> modeTermList = globalControllers.get("global_age_of_onset");
```
Comment (marta-) [inline similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/AbstractPatientSimilarityView.java]: You should probably rename `modeTermList` to something related to "age of onset" to make the code more readable.

File: ui/src/main/resources/PhenoTips/SimilarCases.xml [hunk 2]
```diff
@@ -1588,7 +1644,9 @@
 .match-details table.phenotype-matches .result {
   text-align: left;
 }
-.match-details table.phenotype-matches .phenotypes {
+.match-details table.phenotype-matches .phenotypes,
+.match-details table.mode-of-inheritance .mode-of-inheritance-data,
+.match-details table.age-of-onset .age-of-onset-data {
```
Comment (sdumitriu) [inline ui/src/main/resources/PhenoTips/SimilarCases.xml]: Too many classes, why not add a generic one for all of these tables?

File: ui/src/main/resources/PhenoTips/SimilarCases.xml [hunk 3]
```diff
@@ -619,11 +619,60 @@
...
+      return container;
+    },
+
+    _displayModeOfInheritance : function(r) {
+      var container = new Element('div', {'class' : 'match-type mode-of-inheritance'});
+      container.insert(new Element('h3').update('Mode of inheritance'));
+      var modeOfInheritanceTable = new Element('table', {'class' : 'mode-of-inheritance'});
+      container.insert(modeOfInheritanceTable);
+
+      var row = this._getEmptyTableRow(modeOfInheritanceTable);
```
Comment (veronikaslc) [inline ui/src/main/resources/PhenoTips/SimilarCases.xml]: Lines 636-638 and 653-655 can be one function with parameters

**Comment by itaiGershtansky** [comment]
Addressed comments

**Pull request merged**